### DISCUSSION TITLE
Il multi-tenant va nel template o no?

### DISCUSSION THREAD

**Discussion by matteobadini** [GELLIFY/acme-app#5] [Ideas]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by piccinnigius** [comment]
It's me Pino,
and [here](https://github.com/GELLIFY/acme-supabase/pull/1) you can find my idea of multitenancy made with [Supabase](https://supabase.com/) ❤️

**Reply by GELLIFYBertalan** [reply]
I still believe that multitenancy should not be included as part of the default template. I would prefer to add it only when I need it, rather than having it by default.  

My main reason is the the PR above. As Andrea also mentioned in the team leader channel, if you're unfamiliar with Supabase and don't have some kind of guide or architectural design, it’s not easy to understand how multitenancy is achieved using your example. I would much rather have a practical guide that I can refer to when needed, allowing me to study how it works in detail.

**Reply by matteobadini** [reply]
I tend to agree with @GELLIFYBertalan here. Having **A LOT** of pre-cooked SQL migrations is sub-optimal in the context of the template. Personally, I couldn't understand what was going on

My main concern is the additional complexity that we would introduce trying to solve a problem that could not be there for everyone. I see two possible solutions:

- Either we find a simpler implementation that if not needed it doesn't need to be removed
- We simply document the best way to integrate multitenancy or even better we link an official doc from supabase.

**Reply by dox-op** [reply]
While responding I thought a lot about the core objectives of this project: fast start and overcoming common obstacles.

In this context, multitenancy is certainly worth considering, as it is a feature that many projects may require once they reach their initial distribution phase. I would definitely keep it as an opt-in option. But what's the point of not including it?

The argument I read seems to focus more on familiarity with the tool rather than the actual capabilities it provides from day one. I fully support the idea of finding tools that are easier to adopt and share, without requiring months or weeks of study before they can be effectively used—this aligns well with our project's goals.

As with many other topics we will discuss, I encourage anyone with additional insights to contribute with concrete examples and tools that help reduce the initial learning curve, since nowadays I can't contirbute more on this discussion.
